metabase/metabase
# Migrate SettingsJWTForm to formik

Checking off an item from https://github.com/metabase/metabase/issues/26178

## Summary

1. Converted `SecretKeyWidget` to `FormSecretKey` in `metabase/forms`, which is just a copy of `FormTextInput` with a generate key button and confirmation modal.
2. Added a whitelist of setting keys.
6. Migrate all form fields in `SettingsJWTForm` to our `metabase/forms` components.

before | after
---|---
<img width="767" alt="settings-jwt-original" src="https://github.com/metabase/metabase/assets/116838/48e1639f-446f-4e3d-8795-0e60f9d1b67a"> | <img width="581" alt="CleanShot 2023-11-21 at 17 13 01@2x" src="https://github.com/metabase/metabase/assets/116838/26bcc87c-8a6b-481a-9f88-972eb938a30f">


## How to verify

1. Go to http://localhost:3000/admin/settings/authentication/jwt
2. Fill in random values, and confirm they’re still there on reload.

Run EE tests for `jwt.cy.spec.js` and unit tests.
